MoviesTVSentiments/app/src/test/java/com/google/moviestvsentiments/service/web/NetworkBoundResourceTest.java

```diff
@@ -0,0 +1,170 @@
+import com.google.moviestvsentiments.util.LiveDataTestUtil;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import retrofit2.Response;
+
+@RunWith(JUnitParamsRunner.class)
+public class NetworkBoundResourceTest {
+
+    private static final String LOCAL_VALUE = "Local Value";
+    private static final String SERVER_VALUE = "Server Value";
+
+    @Rule
+    public InstantTaskExecutorRule instantExecutorRule = new InstantTaskExecutorRule();
+
+    private Object[] invokesCorrectMethodsParameters() {
+        return new Object[] {
+            new Object[] {false, new boolean[] {false, true, true, false} },
+            new Object[] {true, new boolean[] {true, true, true, true} }
+        };
+    }
+
+    @Test
+    @Parameters(method = "invokesCorrectMethodsParameters")
+    public void shouldFetchResult_invokesCorrectMethods(boolean shouldFetch,
+                                                        boolean[] expectedInvocations) {
+        final boolean[] invocations = {false, false, false, false};
+        NetworkBoundResource resource = new NetworkBoundResource<String, String>() {
+            @Override protected void saveCallResult(String item) {
+                invocations[0] = true;
+            }
+
+            @Override protected boolean shouldFetch(String data) {
+                invocations[1] = true;
+                return shouldFetch;
+            }
+
+            @Override protected LiveData<String> loadFromRoom() {
+                invocations[2] = true;
+                return new MutableLiveData<>("testValue");
+            }
+
+            @Override protected LiveData<ApiResponse<String>> performNetworkCall() {
+                invocations[3] = true;
+                return new MutableLiveData<>(new ApiResponse(Response.success("Server value")));
+            }
+        };
+
+        resource.getResult().observeForever(data -> {});
+
+        assertThat(invocations).isEqualTo(expectedInvocations);
+    }
+
+    @Test
+    public void networkBoundResource_passesCorrectParameters() {
+        NetworkBoundResource resource = new NetworkBoundResource<String, String>() {
+            @Override protected void saveCallResult(String item) {
+                assertThat(item).isEqualTo(SERVER_VALUE);
+            }
+
+            @Override protected boolean shouldFetch(String data) {
+                assertThat(data).isEqualTo(LOCAL_VALUE);
+                return true;
+            }
+
+            @Override protected LiveData<String> loadFromRoom() {
+                return new MutableLiveData<>(LOCAL_VALUE);
+            }
+
+            @Override protected LiveData<ApiResponse<String>> performNetworkCall() {
+                return new MutableLiveData<>(new ApiResponse(Response.success(SERVER_VALUE)));
+            }
+        };
+
+        resource.getResult().observeForever(data -> {});
```

Comment:
Is this test missing assert statements?

Reply:
No, the asserts are actually in the NetworkBoundResource methods for this tests. I'll move them so the test is more clear.